cmd/run.go

```diff
@@ -299,8 +299,7 @@ func buildGraph(fileSystem *fs.FileSystem, modelFilters []string) *fs.Graph {
 	pb.Increment()
 
 	if graph.Len() == 0 {
-		fmt.Printf("❌ Empty DAG generated for model filter: %s\n", strings.Join(modelFilters, ", "))
-		os.Exit(1)
```

Comment:
I'd actually say still exit here, but maybe with a different error code - say `2` or `3`.

That way code after this doesn't need to handle a `Len()==0` case and you can listen for that speciifc exit code in the caller and behave differently

Reply:
I don't think that would work for my use case without some extra work. 

My use case is that I want to run our dbt `new-base-event` command in the isolated environment with the project/macro files (but no models). This modification makes it possible to still isolate the project/macro files without worrying about whether any models exist or not.  



> I'd actually say still exit here, but maybe with a different error code - say 2 or 3.

If we exit at this point, the caller will have to find another way to isolate the project/macro files without using the isolate-dag command. 


> That way code after this doesn't need to handle a Len()==0

Can you think of where this might be a difficulty? I'd assumed that because it's an iterable object, the code after would just iterate through the empty graph and do nothing instead of copying the model files & schemas, which is fine, I think?

Reply:
DDBT only isolates macro's used by the selected models, no selected models means no isolated macro's

Fair comment on the project files, that could be handled indepdantly. But maybe the solution here is to actually add a parameter to the buildGraph function to not error on that case, and only set that parameter to true for the isoalte-dag command, and leave it false in all other instances.

Reply:
How about re-using the existing `fail-on-not-found` flag à la my latest commit?

Reply:
> DDBT only isolates macro's used by the selected models, no selected models means no isolated macro's

Also I don;t think this is true anymore as there were issues with ddbt not finding macros used by materializations:
https://github.com/monzo/ddbt/blob/master/cmd/isolateDAG.go#L149